### ISSUE TITLE
Implement Local Training Loop (Client)

### ISSUE THREAD

**Issue by IsharaEkanayaka** [cepdnaclk/e20-4yp-backdoor-resilient-federated-nids#6]
Assignee: Member B (Red Team)

Description: Create the Client class in src/client/client.py. It should take a global model, train on local data for E epochs, and return the updated weights.

Acceptance Criteria:

- Client can overfit a small batch of data (Loss goes to 0).
